common/models/user.js

```diff
@@ -1270,9 +1270,13 @@ module.exports = function(User) {
 
   // Access token to normalize email credentials
   User.observe('access', function normalizeEmailCase(ctx, next) {
-    if (!ctx.Model.settings.caseSensitiveEmail && ctx.query.where &&
-        ctx.query.where.email && typeof(ctx.query.where.email) === 'string') {
-      ctx.query.where.email = ctx.query.where.email.toLowerCase();
+    if (!ctx.Model.settings.caseSensitiveEmail && ctx.query.where) {
+      if (typeof ctx.query.where.email === 'string') {
+        ctx.query.where.email = ctx.query.where.email.toLowerCase();
+      } else if (ctx.query.where.email !== null && ctx.query.where.email instanceof Object &&
```

Comment:
There might be other operators in addition to `like`. Do you want to handle them all? For example, `{inq: ['x@Y.com', 'a@B.com']}`

Reply:
Ahh that's very true, I had not considered that initially. This definitely opens up a large can of worms though. I think it would be correct to handle all possible cases to prevent unintended behaviour as is currently exhibited. We would have to account for all possible combinations of operators through some sort of recursive function, and there's also the possibility of different types of patterns which as I previously mentioned should possibly not be modified due to case sensitivity of special regular expression operators or character sets. It seems like there's also different ways to specify patterns and regular expressions which would also need to be detected and accounted for. I have some ideas for going about doing this, but this is not a simple problem. Do you have any thoughts or suggestions for traversing and modifying a filter? I'm not sure if anything similar to this might already exist or not in the Loopback framework. This is something that would also need to be maintained as new operators are added.